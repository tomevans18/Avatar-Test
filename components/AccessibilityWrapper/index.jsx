import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Content from './index.style';

class Index extends Component {
  static propTypes = {
    children: PropTypes.oneOfType([
      PropTypes.arrayOf(PropTypes.node),
      PropTypes.node,
      PropTypes.string,
    ]).isRequired,
  };

  state = {
    tabbing: false,
  };

  componentDidMount() {
    window.addEventListener('keydown', this.handleFirstTab.bind(this));
  }

  componentWillUnmount() {
    window.removeEventListener('keydown', this.handleFirstTab.bind(this));
    window.removeEventListener('mousedown', this.handleMouseDownOnce.bind(this));
  }

  handleFirstTab = e => {
    if (e.keyCode === 9) {
      this.setState({
        tabbing: true,
      });

      window.removeEventListener('keydown', this.handleFirstTab.bind(this));
      window.addEventListener('mousedown', this.handleMouseDownOnce.bind(this));
    }
  };

  handleMouseDownOnce = () => {
    this.setState({
      tabbing: false,
    });

    window.removeEventListener('mousedown', this.handleMouseDownOnce.bind(this));
    window.addEventListener('keydown', this.handleFirstTab.bind(this));
  };

  render() {
    const { children } = this.props;
    const { tabbing } = this.state;
    return <Content tabbing={tabbing}>{children}</Content>;
  }
}

export default Index;
