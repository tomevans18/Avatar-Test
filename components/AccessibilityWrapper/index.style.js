import styled from 'styled-components';

export default styled.div`
  display: flex;
  flex-direction: column;
  flex: 1;

  *:focus {
    outline: ${p => (p.tabbing ? `${p.theme.color.primary} auto 5px` : '0')};
  }
`;
