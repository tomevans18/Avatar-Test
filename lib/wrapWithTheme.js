import { ThemeConsumer } from 'styled-components';
import theme from '../styles/theme';

export default (method, children, options, themeObject = theme) => {
  // eslint-disable-next-line no-underscore-dangle
  ThemeConsumer._currentValue = themeObject;
  return method(children, options);
};
