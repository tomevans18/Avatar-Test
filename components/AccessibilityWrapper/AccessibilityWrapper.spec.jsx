import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import AccessibilityWrapper from './index';

import wrapWithTheme from '../../lib/wrapWithTheme';

const mockRemoveEventListener = jest.fn();
global.removeEventListener = mockRemoveEventListener;

const MockComponent = () => <div />;

describe('Accessibility Wrapper', () => {
  it('should initially render as expected', () => {
    const wrapper = mount(
      <AccessibilityWrapper>
        <MockComponent />
      </AccessibilityWrapper>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.state('tabbing')).toBeFalsy();
    expect(wrapper.prop('tabbing')).toBeFalsy();
  });

  it('should update on tab key press', () => {
    const wrapper = wrapWithTheme(
      mount,
      <AccessibilityWrapper>
        <MockComponent />
      </AccessibilityWrapper>
    );

    wrapper
      .first()
      .instance()
      .handleFirstTab({
        keyCode: 9,
      });

    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.state('tabbing')).toBeTruthy();
  });

  it('should not update on non tab key press', () => {
    const wrapper = wrapWithTheme(
      mount,
      <AccessibilityWrapper>
        <MockComponent />
      </AccessibilityWrapper>
    );

    wrapper
      .first()
      .instance()
      .handleFirstTab({
        keyCode: 12,
      });

    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.state('tabbing')).toBeFalsy();
  });

  it('should update on mouse click', () => {
    const wrapper = wrapWithTheme(
      mount,
      <AccessibilityWrapper>
        <MockComponent />
      </AccessibilityWrapper>
    );

    wrapper.first().setState({
      tabbing: true,
    });

    expect(wrapper.state('tabbing')).toBeTruthy();

    wrapper
      .first()
      .instance()
      .handleMouseDownOnce();

    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.state('tabbing')).toBeFalsy();
  });

  it('should remove event listen on unmount', () => {
    const wrapper = mount(
      <AccessibilityWrapper>
        <MockComponent />
      </AccessibilityWrapper>
    );
    wrapper.unmount();
    global.expect(global.removeEventListener).toHaveBeenCalled();
  });
});
