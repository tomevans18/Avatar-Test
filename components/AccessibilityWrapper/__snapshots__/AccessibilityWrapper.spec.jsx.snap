// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accessibility Wrapper should initially render as expected 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c0 *:focus {
  outline: 0;
}

<Index>
  <indexstyle
    tabbing={false}
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "indexstyle-sc-64rk59-0",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "display:flex;flex-direction:column;flex:1;*:focus{outline:",
              [Function],
              ";}",
            ],
          },
          "displayName": "indexstyle",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "indexstyle-sc-64rk59-0",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      tabbing={false}
    >
      <div
        className="c0"
      >
        <MockComponent>
          <div />
        </MockComponent>
      </div>
    </StyledComponent>
  </indexstyle>
</Index>
`;

exports[`Accessibility Wrapper should not update on non tab key press 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c0 *:focus {
  outline: 0;
}

<Index>
  <indexstyle
    tabbing={false}
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "indexstyle-sc-64rk59-0",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "display:flex;flex-direction:column;flex:1;*:focus{outline:",
              [Function],
              ";}",
            ],
          },
          "displayName": "indexstyle",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "indexstyle-sc-64rk59-0",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      tabbing={false}
    >
      <div
        className="c0"
      >
        <MockComponent>
          <div />
        </MockComponent>
      </div>
    </StyledComponent>
  </indexstyle>
</Index>
`;

exports[`Accessibility Wrapper should update on mouse click 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c0 *:focus {
  outline: undefined auto 5px;
}

<Index>
  <indexstyle
    tabbing={true}
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "indexstyle-sc-64rk59-0",
            "isStatic": false,
            "lastClassName": "gfNcBn",
            "rules": Array [
              "display:flex;flex-direction:column;flex:1;*:focus{outline:",
              [Function],
              ";}",
            ],
          },
          "displayName": "indexstyle",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "indexstyle-sc-64rk59-0",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      tabbing={true}
    >
      <div
        className="c0"
      >
        <MockComponent>
          <div />
        </MockComponent>
      </div>
    </StyledComponent>
  </indexstyle>
</Index>
`;

exports[`Accessibility Wrapper should update on tab key press 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c0 *:focus {
  outline: 0;
}

<Index>
  <indexstyle
    tabbing={false}
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "indexstyle-sc-64rk59-0",
            "isStatic": false,
            "lastClassName": "bnReAS",
            "rules": Array [
              "display:flex;flex-direction:column;flex:1;*:focus{outline:",
              [Function],
              ";}",
            ],
          },
          "displayName": "indexstyle",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "indexstyle-sc-64rk59-0",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      tabbing={false}
    >
      <div
        className="c0"
      >
        <MockComponent>
          <div />
        </MockComponent>
      </div>
    </StyledComponent>
  </indexstyle>
</Index>
`;
